core/src/test/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcherTest.java

```diff
@@ -47,17 +47,22 @@ public long getLongSample() {
         }
     }
 
-    @Test(expected = Exception.class)
-    public void testBadAttributeName() throws Exception {
-        MBeanServer server = ManagementFactory.getPlatformMBeanServer();
-        ObjectName mxbeanName = new ObjectName("JSDKTests:type=TestStub3");
-        TestStub testStub = new TestStub(1, 2.0, 3L);
-        server.registerMBean(testStub, mxbeanName);
-        List<JmxAttributeData> attributes = new ArrayList<>();
-        attributes.add(new JmxAttributeData("Int", "WrongNameIntSample"));
-        attributes.add(new JmxAttributeData("Double", "WrongNameDoubleSample"));
-        attributes.add(new JmxAttributeData("Long", "WrongNameLongSample"));
-        JmxDataFetcher.fetch("JSDKTests:type=TestStub3", attributes);
+    @Test
+    public void testBadAttributeName() {
+        try {
+            MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+            ObjectName mxbeanName = new ObjectName("JSDKTests:type=TestStub3");
+            TestStub testStub = new TestStub(1, 2.0, 3L);
+            server.registerMBean(testStub, mxbeanName);
+            List<JmxAttributeData> attributes = new ArrayList<>();
+            attributes.add(new JmxAttributeData("Int", "WrongNameIntSample"));
+            attributes.add(new JmxAttributeData("Double", "WrongNameDoubleSample"));
+            attributes.add(new JmxAttributeData("Long", "WrongNameLongSample"));
+            JmxDataFetcher.fetch("JSDKTests:type=TestStub3", attributes);
+        }
+        catch (Exception e) {
+            Assert.fail("Exception thrown: "+e);
+        }
```

Comment:
same here

Reply:
Same here - not very familiar with the required test structure. Do we need a way to handle the exception here? What is the proper way for the test purposes - remove? change to finally?